packages/kbn-es-archiver/src/lib/indices/monitoring_index.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { inspect } from 'util';
+
+import type { KibanaClient } from '@elastic/elasticsearch/api/kibana';
+import { ToolingLog } from '@kbn/dev-utils';
+import { Stats } from '../stats';
+import { ES_CLIENT_HEADERS } from '../../client_headers';
+import { deleteIndex } from './delete_index';
+
+export async function cleanMonitoringIndices({
+  client,
+  stats,
+  log,
+}: {
+  client: KibanaClient;
+  stats: Stats;
+  log: ToolingLog;
+}) {
+  deleteMonitoringIndices({ client, stats, log });
+}
+
+/**
+ * Deletes all indices that start with `.monitoring`
+ */
+export async function deleteMonitoringIndices({
+  client,
+  stats,
+  log,
+}: {
+  client: KibanaClient;
+  stats: Stats;
+  log: ToolingLog;
+}) {
+  const indexNames = await fetchMonitoringIndices(client);
+  if (!indexNames.length) {
+    return;
+  }
+
+  await client.indices.putSettings(
+    {
+      index: indexNames,
+      body: { settings: { blocks: { read_only: false } } },
+    },
+    {
+      headers: ES_CLIENT_HEADERS,
+    }
+  );
+
+  await deleteIndex({
+    client,
+    stats,
+    index: indexNames,
+    log,
+  });
+
+  return indexNames;
+}
+
+async function fetchMonitoringIndices(client: KibanaClient) {
+  const resp = await client.cat.indices(
+    { index: '.monitoring-*', format: 'json' },
```

Comment:
Rather than hard-coding the `.monitoring-*` index pattern here I'd much prefer that require specifying an index pattern on the CLI and then pass it in, so that all indexes which match that wildcard pattern will be deleted. Thoughts?

Reply:
Yep. Sure could. I don't consider this PR final, hence the draft status. I was basically just trying it out as a stub here. Curious to get your thoughts on https://github.com/elastic/kibana/pull/88986#discussion_r709785826 before proceeding.

Reply:
Per feedback there, I'll plan to update this to be `node scripts/es_archiver.js delete-indices '.monitoring-*'` so we can use it as needed.